lib/infra/infra-stack.ts

```diff
@@ -53,7 +53,8 @@ export interface infraProps extends StackProps{
     readonly mlNodeCount: number,
     readonly dataNodeStorage: number,
     readonly mlNodeStorage: number,
-    readonly jvmSysPropsString?: string
+    readonly jvmSysPropsString?: string,
+    readonly additionalConfig?: string,
```

Comment:
The setting comes from command line, so it will be treated as string and would need additional logic to process it as list and then concatenate with newline to add it to opensearch.yml file. With current format it is straightforward to process. 
As long as users adhere to the format this should not cause any issues. 
@gaiksaya

Reply:
I agree better to use something like an array of key value pair.

Reply:
@prudhvigodithi @gaiksaya Any example how would it look like user passing the values from command line to process it as list?

Reply:
From user perspective it will look the same but when you process it, the typescript should typecheck it. Like how you check datanodes=9 and 9 is an integer. So if a user passes
additionalConfig=blah it should fail because this is not a key value pair. You need to take an input and process it as key value pair. An example of that is https://github.com/opensearch-project/opensearch-ci/blob/main/lib/compute/env-config.ts
The env class is converting string to key value and pushing it into array.

Reply:
Something like the following
```
--additionalConfig={"plugins.security.nodes_dn": ["CN=*.example.com, OU=SSL, O=Test, L=Test, C=DE", "CN=node.other.com, OU=SSL, O=Test, L=Test, C=DE"], "plugins.security.nodes_dn_dynamic_config_enabled": false}
```
Note the above works for both key with multiple values and single value as well.

Then in typescript we can just use:
```
const doc = new YAML.Document();
doc.contents = jsonObject; // The jsonObject is the additionalConfig input
```

Reply:
@prudhvigodithi @gaiksaya 
What do you think is better user experience
user passing value as additionalConfig=opensearch.experimental.feature.replication_type.enabled:true,cluster.indices.replication.strategy:'SEGMENT'
Or
additionalConfig="{opensearch.experimental.feature.replication_type.enabled:true,cluster.indices.replication.strategy:'SEGMENT'}"

`"` are mandatory when passing as an object from command line
If you miss the ", it is not being read completely. 

Do we want to force user to pass it as a json object and throw an exception they don't follow?

Reply:
Update code to take in additional config as json parameter.
Sample input `--context additionalConfig='{"plugins.security.nodes_dn": ["CN=*.example.com, OU=SSL, O=Test, L=Test, C=DE", "CN=node.other.com, OU=SSL, O=Test, L=Test, C=DE"], "plugins.security.nodes_dn_dynamic_config_enabled": false}'`

corresponding `jsonObj` in code:
```
{
  'plugins.security.nodes_dn': [
    'CN=*.example.com, OU=SSL, O=Test, L=Test, C=DE',
    'CN=node.other.com, OU=SSL, O=Test, L=Test, C=DE'
  ],
  'plugins.security.nodes_dn_dynamic_config_enabled': false
}
```

Corresponding `ymlObj` in code:
```
plugins.security.nodes_dn:
  - CN=*.example.com, OU=SSL, O=Test, L=Test, C=DE
  - CN=node.other.com, OU=SSL, O=Test, L=Test, C=DE
plugins.security.nodes_dn_dynamic_config_enabled: false
```
Please note that the `type of` of ymlObj is `string`. `JSON.parse()` would do the input validation and throw error if the format is wrong. 
@prudhvigodithi @gaiksaya

Reply:
Is it possible to add unit tests to make sure this works as expected? For both valid and invalid yml?